deno_dist/hono.ts

```diff
@@ -83,7 +83,8 @@ export class Hono<E extends Env = Env, S = {}> extends defineDynamicClass()<E, S
             this.addRoute(method, this.path, handler)
           }
         })
-        return this
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return this as any
```

Comment:
It's difficult to avoid type mismatch in this section, so using `any`, but it's OK because this is the implementation part.